Add documentation for delete protection for instances & volumes (#433)
apache/cloudstack-documentation: source/adminguide/storage.rst

upon reboot. See `“Creating a New Compute
Offering” <service_offerings.html#creating-a-new-compute-offering>`_.

Volume delete protection
~~~~~~~~~~~~~~~~~~~~~~~~

> CloudStack provides a feature to protect volumes from accidental deletion. When

Reviewer: ```suggestion
CloudStack protects volumes from accidental deletion using a delete protection flag, which is false by default. When
```

upon reboot. See `“Creating a New Compute
Offering” <service_offerings.html#creating-a-new-compute-offering>`_.

Volume delete protection
~~~~~~~~~~~~~~~~~~~~~~~~

CloudStack protects volumes from accidental deletion using a delete protection
flag, which is false by default. When delete protection is enabled for a volume,
it cannot be deleted through the UI or API. It can only be deleted after
removing delete protection from the volume.

Delete protection can be enabled for a volume via updateVirtualMachine API.

.. code:: bash

>    cmk update volume id=<volume id> deletionprotection=true

Reviewer: ```suggestion
   cmk update volume id=<volume id> deleteprotection=true
```

To remove delete protection, use the following command:

.. code:: bash

>    cmk update volume id=<volume id> deletionprotection=false

Reviewer: ```suggestion
   cmk update volume id=<volume id> deleteprotection=false
```
